src/deepsparse/utils/onnx.py

```diff
@@ -264,3 +267,78 @@ def override_onnx_input_shapes(
     finally:
         os.unlink(shaped_model.name)
         shaped_model.close()
+
+
+def truncate_onnx_model(
+    onnx_filepath: str,
+    output_filepath: str,
+    final_node_names: List[str],
+    graph_output_names: List[str],
+    graph_output_shapes: Optional[List[List[int]]] = None,
+) -> None:
+    """
+    :param onnx_filepath: file path to onnx model
+    :param output_filepath: file path to save new onnx model
+    :param final_node_names: list of node names whose outputs will become the
+        outputs of the graph
+    :param graph_output_names: list of names to call the graph outputs. Names
+        correspond with the outputs specified in final_node_names
+    :param graph_output_types: list of numpy dtypes
+    :param graph_output_shapes: list of shapes for each output. If not provided,
+        defaults to [None] for each output and leads to slight performance loss
+    :return: None
+    """
+    if graph_output_shapes is None:
+        graph_output_shapes = [[None]] * len(final_node_names)
+
+    if len(final_node_names) != len(graph_output_names) != len(graph_output_shapes):
+        raise ValueError(
+            f"length of final_node_names {len(final_node_names)}, "
+            f"graph_output_names {len(graph_output_names)}, and "
+            f"graph_output_shapes {len(graph_output_shapes)} must all match"
+        )
+
+    if len(set(final_node_names)) != len(final_node_names):
+        raise ValueError("final_node_names must not contain duplicate names")
+
+    if len(set(graph_output_names)) != len(graph_output_names):
+        raise ValueError("graph_output_names must not contain duplicate names")
+
+    model = onnx.load(onnx_filepath)
+    final_nodes = [node for node in model.graph.node if node.name in final_node_names]
+
+    if len(final_nodes) != len(final_node_names):
+        raise ValueError("Could not find final node names in model graph")
+
+    for i, (final_node, graph_output_name, graph_output_shape) in enumerate(
```

Comment:
remove `i` + `enumerate` if not used. additionally prefer to avoid single character names